packages/block-library/src/query/edit/query-content.js

```diff
@@ -152,6 +154,24 @@ export default function QueryContent( {
 				setAttributes={ setAttributes }
 				clientId={ clientId }
 			/>
+			{ hasPatterns && (
+				<InspectorControls
+					group={ isNavigationMode ? 'navigationMode' : 'default' }
```

Comment:
Maybe we shouldn't need this bifurcation and the block editor should know how to handle write mode? The more I say it the less I think it's doable. Even here, query block shows a design picker, navigation block shows a tree of inner blocks.